packages/material-ui/src/TabScrollButton/TabScrollButton.js

```diff
@@ -2,49 +2,78 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import KeyboardArrowLeft from '../internal/svg-icons/KeyboardArrowLeft';
 import KeyboardArrowRight from '../internal/svg-icons/KeyboardArrowRight';
-import withStyles from '../styles/withStyles';
 import ButtonBase from '../ButtonBase';
 
-export const styles = {
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import { getTabScrollButtonUtilityClass } from './tabScrollButtonClasses';
+
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...styles[styleProps.orientation],
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, orientation, disabled } = styleProps;
+
+  const slots = {
+    root: ['root', orientation, disabled && 'disabled'],
+  };
+
+  return composeClasses(slots, getTabScrollButtonUtilityClass, classes);
+};
+
+const TabScrollButtonRoot = experimentalStyled(
+  ButtonBase,
+  {},
+  {
+    name: 'MuiTabScrollButton',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(({ styleProps }) => ({
   /* Styles applied to the root element. */
-  root: {
-    width: 40,
-    flexShrink: 0,
-    opacity: 0.8,
-    '&$disabled': {
-      opacity: 0,
-    },
+  width: 40,
+  flexShrink: 0,
+  opacity: 0.8,
+  '&.Mui-disabled': {
+    opacity: 0,
   },
   /* Styles applied to the root element if `orientation="vertical"`. */
-  vertical: {
+  ...(styleProps.orientation === 'vertical' && {
     width: '100%',
     height: 40,
     '& svg': {
       transform: 'rotate(90deg)',
     },
-  },
-  /* Pseudo-class applied to the root element if `disabled={true}`. */
-  disabled: {},
-};
+  }),
+}));
+
+const TabScrollButton = React.forwardRef(function TabScrollButton(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiTabScrollButton' });
+  const { className: classNameProp, direction, orientation, disabled, ...other } = props;
 
-const TabScrollButton = React.forwardRef(function TabScrollButton(props, ref) {
-  const { classes, className: classNameProp, direction, orientation, disabled, ...other } = props;
+  const styleProps = {
+    ...props,
+    orientation,
+  };
```

Comment:
```suggestion
  // TODO: convert to simple assignment after the type error in defaultPropsHandler.js:60:6 is fixed
  const styleProps = { ...props };
```

`orientation` doesn't have default value, no need to be specified here.